client/components/login/login.js

```diff
@@ -206,27 +195,25 @@ export default class Login extends React.Component {
       }),
     })
       .then((res = {}) => {
-        if (res.data) setIsActive(res.data.is_active);
-        return handleAuthentication(
-          settings.mobile_phone_verification,
-          res.data,
-        );
+        return handleAuthentication(res.data);
       })
       .catch((error) => {
         const {data} = error.response;
-        setIsActive(data.is_active);
```

Comment:
here, it could be called here. Why did you opt for not doing this and instead call `setUserData()` both in `handleAuthentication` and then below? Isn't this redundant?

Keep in mind redundancy like this is what we have to get rid of. We need to simplify the code and make it easier to deal with. These minor details are not helping us in achieving the goal.

I want to hear your justification for such code changes before deciding what to do with them.